src/components/Heading/Heading.tsx

```diff
@@ -13,34 +13,47 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from '@emotion/styled';
+import { FC, HTMLProps } from 'react';
 import { css } from '@emotion/core';
 import isPropValid from '@emotion/is-prop-valid';
 
-import { childrenPropType } from '../../util/shared-prop-types';
-import { sizes } from '../../styles/constants';
+import styled, { StyleProps } from '../../styles/styled';
 
-const { KILO, MEGA, GIGA, TERA, PETA, EXA, ZETTA } = sizes;
+type Size = 'kilo' | 'mega' | 'giga' | 'tera' | 'peta' | 'exa' | 'zetta';
 
-const mobileSizeMap = {
-  [KILO]: KILO,
-  [MEGA]: MEGA,
-  [GIGA]: MEGA,
-  [TERA]: GIGA,
-  [PETA]: TERA,
-  [EXA]: PETA,
-  [ZETTA]: PETA
+export interface HeadingProps
+  extends Omit<HTMLProps<HTMLHeadingElement>, 'size'> {
```

Comment:
Can you add the HTML "base" props to the Text component as well please? You can probably extend the generic `HTMLElement` type:

```ts
export TextProps extends Omit<HTMLProps<HTMLElement>, 'size'> {...}
```

Reply:
**Edit:** Looks like you have to use `HTMLParagraphElement` to make TypeScript happy.

Reply:
Ugh, and it's leading to more errors in other components :(

If you don't wanna deal with it now, leave it and I'll tackle it later.